test/ApiService.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (c) 2018, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const EOL = require('os').EOL;
+
+const async = require('async');
+const isempty = require('lodash.isempty');
+const request = require('request');
+
+let authTokenMapi;
+const baasAuthTokens = {};
+
+function getBaseUrl(isBaasRequest) {
+  if (isBaasRequest && process.env.KINVEY_CLI_BAAS) {
+    return process.env.KINVEY_CLI_BAAS;
+  }
+
+  let instance = process.env.KINVEY_CLI_INSTANCE;
+  if (instance && instance.includes('localhost')) {
+    return instance;
+  }
+
+  if (!instance) {
+    instance = 'kvy-us1';
+  }
+
+  return isBaasRequest ? `https://${instance}-baas.kinvey.com` : `https://${instance}-manage.kinvey.com`;
```

Comment:
Is _getBaseUrl_ working on the local dev machine - isn't the MAPI Url something like http://localhost:8888?